tools/cldr-code/src/main/java/org/unicode/cldr/util/LocalesTxtReader.java

```diff
@@ -0,0 +1,205 @@
+package org.unicode.cldr.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import com.google.common.collect.ImmutableSet;
+import com.ibm.icu.impl.Relation;
+import com.ibm.icu.util.ICUUncheckedIOException;
+
+public class LocalesTxtReader {
+    Map<Organization, Map<String, Level>> platform_locale_level = null;
+    Map<Organization, Relation<Level, String>> platform_level_locale = null;
+    Map<String, Map<String, String>> platform_locale_levelString = null;
+    Map<Organization, Map<String, Integer>> organization_locale_weight = null;
+    Map<Organization, Map<String, Set<String>>> organization_locale_match = null;
+
+    public static final String DEFAULT_NAME = "Locales.txt";
+
+    public LocalesTxtReader() {
+    }
+
+    /**
+     * Read from Locales.txt, from the default location
+     * @param lstreg stream to read from
+     */
+    public LocalesTxtReader loadPlatformLocaleStatus(StandardCodes sc) {
```

Comment:
I don't see what is going on here. It looks this will be performed multiple times.

Reply:
No, only once. But I'll rename the function in this class to `read()` to make it clearer.